analyzers/src/SonarAnalyzer.CSharp/Extensions/MethodDeclarationSyntaxExtensions.cs

```diff
@@ -17,13 +17,46 @@
  * along with this program; if not, write to the Free Software Foundation,
  * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  */
+
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
 using Microsoft.CodeAnalysis.CSharp.Syntax;
+using StyleCop.Analyzers.Lightup;
 
 namespace SonarAnalyzer.Extensions
 {
     internal static class MethodDeclarationSyntaxExtensions
     {
+        /// <summary>
+        /// Returns true if the method throws exceptions or returns null.
+        /// </summary>
+        public static bool ThrowsOrReturnsNull(this MethodDeclarationSyntax syntaxNode) =>
+            syntaxNode.DescendantNodes().OfType<ThrowStatementSyntax>().Any() ||
+            syntaxNode.DescendantNodes().OfType<ExpressionSyntax>().Any(expression => expression.IsKind(SyntaxKindEx.ThrowExpression)) ||
+            syntaxNode.DescendantNodes().OfType<ReturnStatementSyntax>().Any(returnStatement => returnStatement.Expression.IsKind(SyntaxKind.NullLiteralExpression)) ||
+            // For simplicity this returns true for any method witch contains a NullLiteralExpression but this could be a source of FNs
+            syntaxNode.DescendantNodes().OfType<ExpressionSyntax>().Any(expression => expression.IsKind(SyntaxKind.NullLiteralExpression));
```

Comment:
In case anyone is interested in optimizing this method. It iterates the whole tree of the method a couple of times. This could be done in one go about like this:
```cs
syntaxNode.DescendantNodes().Any(x => x.IsAnyKind(SyntaxKind.ThrowStatement, SyntaxKindEx.ThrowExpression, SyntaxKind.NullLiteralExpression))
```

Reply:
`// For simplicity this returns true for any method witch contains a NullLiteralExpression but this could be a source of FNs`

I think this is just wrong. Is someone interested in looking into it?

Reply:
I will take a look at the PR right after this one. 
It would be good to merge this PR today so that peach is ok tomorrow as we also blocking the security bubble if I've understood correctly.

Reply:
Nice, thanks.